tests/containers/bci_prepare.pm

```diff
@@ -24,74 +24,62 @@ use warnings;
 use Mojo::Base qw(consoletest);
 use XML::LibXML;
 use utils qw(zypper_call script_retry);
-use version_utils qw(get_os_release is_sle is_opensuse);
+use version_utils qw(get_os_release);
 use db_utils qw(push_image_data_to_db);
 use containers::common;
 use testapi;
 use serial_terminal 'select_serial_terminal';
 
-sub packages_to_install {
+sub prepare_virtual_env {
     my ($version, $sp, $host_distri) = @_;
     my $arch = get_required_var('ARCH');
     my $scc_timeout = 1200;    # SCC can take really long timetimes
+    my $install_timeout = 600;
+    my $virtualenv = 'bci/bin/activate';
+    my $python = 'python3';
 
-    my $bci_virtualenv = get_var('BCI_VIRTUALENV', 0);
-
-    # Avoid PackageKit to conflict about lock with zypper
-    script_run("timeout 20 pkcon quit") if (is_sle || is_opensuse);
+    record_info('Install', 'Install needed packages');
 
-    # common packages
-    my @packages = ('git-core', 'python3', 'jq');
-    if ($host_distri eq 'ubuntu') {
-        push @packages, ('python3-pip');
-        push @packages, ('python3-virtualenv') if ($bci_virtualenv);
-    } elsif ($host_distri eq 'rhel' && $version > 7) {
-        push @packages, ('python3-pip');
-        push @packages, ('python3-virtualenv') if ($bci_virtualenv);
+    if ($host_distri =~ /ubuntu/) {
+        # Sometimes, the host doesn't get an IP automatically via dhcp, we need force it just in case
+        assert_script_run("dhclient -v");
+        # This command prevents a prompt that asks for services to be restarted
+        # causing a delay of 5min on each package install
+        script_run('export DEBIAN_FRONTEND=noninteractive');
+        script_retry("apt-get -y install python3.12-venv", timeout => $install_timeout);
```

Comment:
is possible install this without hardcoded version ?  I othervise it will need more care in future with ubuntu new versions ...

Reply:
oh I thought I used `python3-venv` . yeah I will try that